reference.md

```diff
@@ -0,0 +1,1393 @@
+    [...]
+    Processing triggers for sgml-base (1.26+nmu4ubuntu1) ...
+    Downloading parts list|
+    Preparing to pull part3
+    Pulling part3
+    Preparing to pull part2
+    Pulling part2
+    Preparing to pull part1
+    Pulling part1
+    Preparing to build part3
+    Building part3
+    Preparing to build part2
+    Building part2
+    Preparing to build part1
+    Building part1
+    Staging part3
+    Staging part2
+    Staging part1
+    Priming part3
+    Priming part2
+    Priming part1
+    Snapping 'my-snap' |
+    Snapped my-snap_0_amd64.snap
+    Retrieved my-snap_0_amd64.snap
+
+### Full lifecycle in a real project
+
+    $ snapcraft/tour/00-SNAPCRAFT/01-easy-start$ snapcraft
+    Preparing to pull gnu-hello
+    Pulling gnu-hello
+    Downloading 'hello-2.10.tar.gz'[=====================================================] 100%
+    Preparing to build gnu-hello
+    Building gnu-hello
+    ./configure --prefix=
+    checking for a BSD-compatible install... /usr/bin/install -c
+    [...]
+    make -j1
+    [...]
+    make[1]: Leaving directory '/home/elopio/workspace/canonical/snapcraft/tour/00-SNAPCRAFT/01-easy-start/parts/gnu-hello/build'
+    Staging gnu-hello
+    Priming gnu-hello
+    Snapping 'hello' |
+    Snapped hello_2.10_amd64.snap
+
+### Errors
+
+Try to run a command that doesn't exist:
+
+    $ snapcraft idontexist
+    Usage:
+      snapcraft [options] [--enable-geoip --no-parallel-build]
+      snapcraft [options] init
+      snapcraft [options] pull [<part> ...]  [--enable-geoip]
+      snapcraft [options] build [<part> ...] [--no-parallel-build]
+      snapcraft [options] stage [<part> ...]
+      snapcraft [options] prime [<part> ...]
+      snapcraft [options] strip [<part> ...]
+      snapcraft [options] clean [<part> ...] [--step <step>]
+      snapcraft [options] snap [<directory> --output <snap-file>]
+      snapcraft [options] cleanbuild
+      snapcraft [options] login
+      snapcraft [options] logout
+      snapcraft [options] register <snap-name>
+      snapcraft [options] upload <snap-file>
+      snapcraft [options] list-plugins
+      snapcraft [options] tour [<directory>]
+      snapcraft [options] update
+      snapcraft [options] define <part-name>
+      snapcraft [options] search [<query> ...]
+      snapcraft [options] help (topics | <plugin> | <topic>) [--devel]
+      snapcraft (-h | --help)
+      snapcraft --version
+
+Try to show help for an unexisting topic:
+
+    $ snapcraft help idontexist
+    The plugin does not exist. Run `snapcraft list-plugins` to see the available plugins.
+
+Try a lifecycle command without a yaml:
+
+    $ snapcraft
+    Could not find snapcraft.yaml.  Are you sure you are in the right directory?
+    To start a new project, use 'snapcraft init'
+
+Try to init in a directory with a snapcraft.yaml:
+
+    $ snapcraft init
+    snapcraft.yaml already exists!
```

Comment:
This should fail also if .snapcraft.yaml exists.
https://bugs.launchpad.net/snapcraft/+bug/1604326